src/streamlink/plugins/drdk.py

```diff
@@ -1,135 +1,75 @@
-                                ],
-                                validate.get(0)
-                            )
-                        },
-                        validate.get("Streams")
-                    )
-                ],
-                "Server": validate.text
-            }],
-            validate.filter(lambda s: s["LinkType"] in STREAMING_TYPES)
-        )
-    },
-    validate.get("StreamingServers", {})
-)
-
-_video_schema = validate.Schema(
-    {"Data": [{
-        "Assets": validate.all(
-            [{validate.optional("Links"): validate.all(
-                [{
-                    "Target": validate.text,
-                    "Uri": validate.text
-                }],
-                validate.filter(lambda l: l["Target"] in STREAMING_TYPES)
-            )}],
-            validate.filter(lambda a: "Links" in a)
-        )
-    }]},
-    validate.get("Data", {}),
-    validate.get(0, {}),
-    validate.get("Assets", {}),
-    validate.get(0, {}),
-    validate.get("Links", []),
-)
+    url_re = re.compile(r'''
+        https?://(?:www\.)?dr\.dk/drtv
+        (/(kanal)/[\w-]+)
+    ''', re.VERBOSE)
+
+    _live_data_schema = validate.Schema(
+        {'item': {'customFields': {
+            'hlsURL': validate.url(),
+            'hlsWithSubtitlesURL': validate.url(),
+        }}},
+        validate.get('item'),
+        validate.get('customFields'),
+    )
 
+    arguments = PluginArguments(
+        PluginArgument(
+            'live-subtitles',
+            action='store_true',
+            help="""
+            Enable Danish subtitles for live channels if they are available.
+            """,
+        ),
+    )
 
-class DRDK(Plugin):
     @classmethod
     def can_handle_url(cls, url):
-        return _url_re.match(url)
+        return cls.url_re.match(url) is not None
 
-    def _get_streams(self):
-        match = _url_re.match(self.url)
-        if not match:
-            return
+    def _get_live(self, path):
+        res = self.session.http.get(self.live_api_url.format(path))
+        playlists = self.session.http.json(res, schema=self._live_data_schema)
 
-        if match.group("channel"):
-            return self._get_live_streams(match.group("channel"))
+        if self.get_option('live_subtitles'):
+            selected_playlist = 'hlsWithSubtitlesURL'
         else:
-            return self._get_vod_streams(match.group("program"))
+            selected_playlist = 'hlsURL'
 
-    def _get_vod_streams(self, program):
-        res = self.session.http.get(VOD_API_URL.format(program))
-        video = self.session.http.json(res, schema=_video_schema)
+        for playlist_name, playlist_url in playlists.items():
+            if selected_playlist == playlist_name:
+                log.debug("{0}={1}".format(playlist_name, playlist_url))
+                return HLSStream.parse_variant_playlist(
+                    self.session,
+                    playlist_url,
+                )
```

Comment:
Please don't add a plugin parameter for that. Either just use the subtitles playlist, as I don't think there will be a difference content-wise, or combine the streams of both playlists and give the streams of one playlist different names, if necessary. There also seem to be playlists with an `Eu` suffix. Is there a difference other than maybe server location? Combining those sounds like something that could be done. I think it'd also be good to make those additional keys of the API reponse optional in the scheme and only use them when available.

Alternatively, there also seem to be DASH streams available, but I don't think it's necessary to include them.

I can't access any of the streams though due to 403 errors (geo-blocked).

Reply:
OK on the plugin parameter.  Regarding the ```Eu``` playlists, I'm not sure what the difference is - the content looked to be the same.  I decided not to use them in the code I've submitted simply because they looked somewhat temporary, with hostnames of ```drtesthls``` and ```drtesttexthls```.

Funnily enough, I did initially combine them before in the way I think you're suggesting, but I noticed some things weren't appearing in the final list.  A second look shows these are only the ```_alt``` URLs that are being trimmed though.

I went through my backups to find what I had before.  Using:
```
    _live_data_schema = validate.Schema(
        {'item': {'customFields': {
            validate.optional('hlsURL'): validate.url(),
            validate.optional('hlsWithSubtitlesURL'): validate.url(),
            validate.optional('hlsURLEu'): validate.url(),
            validate.optional('hlsWithSubtitlesURLEu'): validate.url(),
        }}},
        validate.get('item'),
        validate.get('customFields'),
    )

    def _get_live(self, path):
        res = self.session.http.get(self.api_url.format(path))
        play_lists = self.session.http.json(res, schema=self._live_data_schema)

        streams = {}
        for name, url in play_lists.items():
            streams.update(HLSStream.parse_variant_playlist(
                self.session,
                url,
                name_prefix="{0}_".format(name),
            ))

        return streams
```

To get:
```
Available streams: hlswithsubtitlesurleu_720p, hlsurl_360p, hlswithsubtitlesurl_576p, hlswithsubtitlesurl_360p, hlsurl_576p, hlsurleu_432p, hlsurleu_360p, hlsurl_432p, hlswithsubtitlesurleu_360p, hlsurl_720p, hlsurl_270p, hlsurleu_576p, hlswithsubtitlesurl_432p, hlswithsubtitlesurl_720p, hlsurleu_720p, hlswithsubtitlesurl_270p
```

If that looks OK, I'll push it.  Thanks.

Reply:
I think I prefer this, as it brings back the ```worst```, ```best``` and various ```_alt``` options in the available streams list:
```
    def _get_live(self, path):
        res = self.session.http.get(self.live_api_url.format(path))
        play_lists = self.session.http.json(res, schema=self._live_data_schema)

        streams = {}
        for name, url in play_lists.items():
            name_prefix = ''
            if name == 'hlsWithSubtitlesURL':
                name_prefix="{0}_".format('subtitled')

            streams.update(HLSStream.parse_variant_playlist(
                self.session,
                url,
                name_prefix=name_prefix,
            ))

        return streams
```
```
Available streams: subtitled_432p, subtitled_720p, subtitled_432p_alt, subtitled_270p_alt, subtitled_576p, subtitled_360p_alt, subtitled_360p, subtitled_720p_alt, subtitled_270p, subtitled_576p_alt, 270p_alt (worst), 270p, 360p_alt, 360p, 432p_alt, 432p, 576p_alt, 576p, 720p_alt, 720p (best)
```